tiledb/bioimg/converters/scale.py

```diff
@@ -0,0 +1,272 @@
+            )
+
+        with Pool(multiprocessing.cpu_count()) as pool:
+            pool.map(
+                partial(
+                    self._scale_progressive,
+                    base=base,
+                    out_data=out_data,
+                    level=level,
+                    scale_method=method.value,
+                ),
+                iter_tiles(out_data.domain),
+            )
+
+    def _scale(
+        self,
+        tile: Tuple[slice],
+        base: tiledb.array.DenseArray,
+        out_data: tiledb.array.DenseArray,
+        level: int,
+        scale_method: int,
+    ) -> None:
+        scaled_tile = list(tile)
+        slice_shape = []
+        for sl in scaled_tile:
+            slice_shape.append(sl.stop - sl.start)
+
+        scaled_tile[self._x_index] = slice(
+            int(tile[self._x_index].start * self._scale_factor[level]),
+            int(
+                min(
+                    tile[self._x_index].stop * self._scale_factor[level],
+                    base.shape[self._x_index],
+                )
+            ),
+        )
+        scaled_tile[self._y_index] = slice(
+            int(tile[self._y_index].start * self._scale_factor[level]),
+            int(
+                min(
+                    tile[self._y_index].stop * self._scale_factor[level],
+                    base.shape[self._y_index],
+                )
+            ),
+        )
+
+        out_data[tile] = resize(
+            base[tuple(scaled_tile)],
+            tuple(slice_shape),
+            preserve_range=True,
+            order=scale_method,
+            anti_aliasing=True,
+        )
+
+    def _scale_progressive(
+        self,
+        tile: Tuple[slice],
+        base: tiledb.array.DenseArray,
+        out_data: tiledb.array.DenseArray,
+        level: int,
+        scale_method: int,
+    ) -> None:
+        scaled_tile = list(tile)
+        slice_shape = []
+        for sl in scaled_tile:
+            slice_shape.append(sl.stop - sl.start)
+
+        scaled_tile[self._x_index] = slice(
+            int(tile[self._x_index].start * self._scale_factor_progressive[level]),
+            int(
+                min(
+                    tile[self._x_index].stop * self._scale_factor_progressive[level],
+                    base.shape[self._x_index],
+                )
+            ),
+        )
+        scaled_tile[self._y_index] = slice(
+            int(tile[self._y_index].start * self._scale_factor_progressive[level]),
+            int(
+                min(
+                    tile[self._y_index].stop * self._scale_factor_progressive[level],
+                    base.shape[self._y_index],
+                )
+            ),
+        )
+
+        out_data[tile] = resize(
+            base[tuple(scaled_tile)],
+            tuple(slice_shape),
+            preserve_range=True,
+            order=scale_method,
+            anti_aliasing=True,
+        )
```

Comment:
Lots of duplicated (or quadruplicated) code in this module, can be DRY-fied (not necessarily this PR).